src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowEditBase.cs

```diff
@@ -181,34 +181,32 @@ protected WhereClause GetWhereClause(bool parameterize)
                 }
                 else
                 {
-                    if (cellData.RawObject is byte[] ||
-                        col.DbColumn.DataTypeName.Equals("TEXT", StringComparison.OrdinalIgnoreCase) ||
-                        col.DbColumn.DataTypeName.Equals("NTEXT", StringComparison.OrdinalIgnoreCase))
+                    if (parameterize)
                     {
-                        // Special cases for byte[] and TEXT/NTEXT types
-                        cellDataClause = "IS NOT NULL";
-                    }
-                    else
-                    {
-                        // General case is to just use the value from the cell
-                        if (parameterize)
+                        // Add a parameter and parameterized clause component
+                        // NOTE: We include the row ID to make sure the parameter is unique if
+                        //       we execute multiple row edits at once.
+                        string paramName = $"@Param{RowId}{col.Ordinal}";
+                        if (col.DbColumn.DataTypeName.Equals("TEXT", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Why did you get rid of the byte[] type check?

Reply:
I think that using equals should handle the comparison of byte arrays just fine, unless I'm wrong about that.

Reply:
Hm, I'm not sure. I'd suggest making sure to test it out first. varbinary I expect probably works fine so perhaps this logic is for the image type? https://docs.microsoft.com/en-us/sql/t-sql/data-types/ntext-text-and-image-transact-sql?view=sql-server-ver15 Make sure you test that to verify.